index.d.ts

```diff
@@ -0,0 +1,518 @@
+
+declare module 'fuse-bindings' {
+
+	function mount(mountPoint: string, ops: MountOptions,
+		cb: (code: number) => void): void;
+
+	interface MountOptions {
+	}
+
+	// XXX readme on https://github.com/mafintosh/fuse-bindings has all things
+	// XXX https://github.com/bcle/fuse4js/blob/master/example/mirrorFS.js has
+	// fuse4js comments, that may be useful, as these libraries should be
+	// comparable in an overlapping area.
+
+	interface MountOptions {
+		/**
+		 * Set to true to force mount the filesystem (will do an unmount first).
+		 */
+		force?: boolean;
+
+		options?: string[];
+
+		displayFolder?: boolean;
+
+		/**
+		 * Called on filesystem init.
+		 * @param cb 
+		 */
+		init?(cb: (code: number) => void): void;
+
+		/**
+		 * Called before the filesystem accessed a file.
+		 * @param path 
+		 * @param mode 
+		 * @param cb 
+		 */
+		access?(path: string, mode, cb: (code: number) => void): void;
+
+		/**
+		 * Called when the filesystem is being stat'ed.
+		 * @param path 
+		 * @param cb accepts a fs stat object after the return code.
+		 */
+		statfs?(path: string, cb: (code: number, fsStat: FSStat) => void): void;
+
+		/**
+		 * Called when a path is being stat'ed. 
+		 * @param path 
+		 * @param cb accepts a stat object (similar to the one returned in fs.stat
+		 * path, cb)) after the return code.
+		 */
+		getattr?(path: string, cb: (code: number, stats: Stats) => void): void;
```

Comment:
I am seeing two signatures for cb:
*  `(code: number)` [source](https://github.com/mafintosh/fuse-bindings/blob/a5f67aa521af35253fee9ebc3cda9a30603fd1c9/example.js#L41)
*  `(code: 0, stats: Stats)` [source](https://github.com/mafintosh/fuse-bindings/blob/a5f67aa521af35253fee9ebc3cda9a30603fd1c9/example.js#L14)

This can be done by [function overloading](https://stackoverflow.com/a/12689054/5169684) with the above signatures or less explicitly but more succinctly:

```ts
cb: (code: number, stats?: Stats)
```